### DISCUSSION TITLE
Feature Request for GitHub Copilot in Jetbrains Plugin - Commit Generation and Custom Starting Prompts

### DISCUSSION THREAD

**Discussion by mkarolczyk** [community/community#102407] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Dear GitHub Copilot Support Team,

I hope this message finds you well. I am writing to suggest two potential enhancements for the GitHub Copilot plugin, specifically for the Jetbrains suite of development tools. These features, I believe, could significantly augment the efficiency and customization of the development process.

Commit Generation Feature:

Description: The ability for GitHub Copilot to assist in generating commit messages based on the context of code changes. This feature could analyze the code differences and suggest a concise, informative commit message that aligns with best practices.
Benefit: This would streamline the workflow for developers, especially in large projects where commit message consistency is crucial. It would also serve as a learning tool for junior developers to understand the art of writing effective commit messages.
Custom Starting Prompt for Project Conventions:

Description: An option to define a starting prompt or a set of guidelines that GitHub Copilot can use to tailor its suggestions according to specific project conventions or coding styles.
Benefit: This feature would ensure that Copilot's suggestions are not just generic but are aligned with the project's specific requirements, coding styles, and best practices. It would enhance the tool's utility in diverse development environments and promote consistency throughout the project's lifecycle.
These features, in my opinion, would not only enhance the capabilities of GitHub Copilot but also provide a more tailored and efficient experience for developers using Jetbrains tools.

**Comment by jgwconsulting** [comment]
Both excellent ideas. Github Copilot is considerably better than the current JetBrains AI Assistant, particularly for code completion, but I'd really miss commit message generation if I were to replace it.

**Comment by uncle-light** [comment]
Hope to support soon

**Comment by salihkavaf** [comment]
I've tried both Github Copilot and JetBrains AI Assitance on Rider, and I find Copilot to perform better in most scenarios. However, the "commit messages" feature is why I still stick with JetBrains AI.
I hope this feature gets supported soon so we can enjoy Copilot on Rider too!

**Comment by maui1911** [comment]
I would love this feature!

**Comment by lobner** [comment]
While i upvote it... the 

> "It would also serve as a learning tool for junior developers to understand the art of writing effective commit messages."

seems to be just the need this would render obsolete. Right? :)

**Comment by zebra-pro-dev** [comment]
https://www.jetbrains.com/help/idea/work-with-commits.html#customize-commit-message-prompt

here is their official feature something like this

**Comment by MeGaNeKoS** [comment]
A UI for configuring this setting would be ideal, but unfortunately, it hasn't been added yet.

In the meantime, you can adjust the behavior by creating specific files under the `.github` directory. Although the full list of supported filenames hasn't been officially documented, the [release notes for version 1.5.42-241](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/715074) mention two recognized files:

* `.github/copilot-instructions.md`
* `.github/git-commit-instructions.md`

These allow you to define custom instructions for Chat and Git commit message generation.

**Reply by inhanbyeol94** [reply]
Thanks, I was able to solve the problem! But I want these to be applied to the interface
